providers/aws-ec2/src/main/java/org/jclouds/aws/ec2/domain/VPC.java

```diff
@@ -0,0 +1,104 @@
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jclouds.aws.ec2.domain;
+
+import java.util.Map;
+
+import org.jclouds.javax.annotation.Nullable;
+import org.jclouds.json.SerializedNames;
+
+import com.google.auto.value.AutoValue;
+
+/**
+ * Amazon EC2 VPC.
+ *
+ * @see <a href="http://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_Vpc.html" >doc</a>
+ */
+@AutoValue
+public abstract class VPC {
+
+   public enum State {
+      /**
+       * The subnet is available for use.
+       */
+      AVAILABLE,
+      /**
+       * The subnet is not yet available for use.
+       */
+      PENDING, UNRECOGNIZED;
+      public String value() {
+         return name().toLowerCase();
+      }
+
+      public static State fromValue(String v) {
+         try {
+            return valueOf(v.toUpperCase());
+         } catch (IllegalArgumentException e) {
+            return UNRECOGNIZED;
+         }
+      }
+   }
+
+   @Nullable
+   public abstract String id();
+   @Nullable
+   public abstract State state();
+   @Nullable
+   public abstract String cidrBlock();
+   @Nullable
+   public abstract String dhcpOptionsId();
+   @Nullable
+   public abstract String instanceTenancy();
+   @Nullable
+   public abstract Boolean isDefault();
+   @Nullable
+   public abstract Map<String, String> tags();
+
+
+   @SerializedNames({ "vpcId", "state", "cidrBlock", "dhcpOptionsId", "instanceTenancy", "isDefault", "tagSet" })
+   public static VPC create(String id, State state, String cidrBlock, String dhcpOptionsId, String instanceTenancy, Boolean isDefault, Map<String, String> tags) {
+      return builder()
+              .id(id)
+              .state(state)
+              .isDefault(isDefault)
+              .cidrBlock(cidrBlock)
+              .dhcpOptionsId(dhcpOptionsId)
+              .instanceTenancy(instanceTenancy)
+              .tags(tags)
+              .build();   
+   }
+
+   VPC() {}
+
+   public static Builder builder() {
+      return new AutoValue_VPC.Builder();
+   }
+
+   @AutoValue.Builder
+   public abstract static class Builder {
+
+      public abstract Builder id(String id);
+      public abstract Builder state(State state);
+      public abstract Builder cidrBlock(String cidrBlock);
+      public abstract Builder dhcpOptionsId(String dhcpOptionsId);
+      public abstract Builder instanceTenancy(String instanceTenancy);
+      public abstract Builder isDefault(Boolean isDefault);
+      public abstract Builder tags(Map<String, String> tags);
```

Comment:
Oops. Have a look [here](https://github.com/jclouds/jclouds-labs-openstack/blob/f831f59a1df246a9f6c20fc9b458d4ae971ef8ef/openstack-neutron/src/main/java/org/jclouds/openstack/neutron/v2/domain/CreateFirewallPolicy.java#L127-L139). The idea is to provide a hidden build method, which is the one Google Auto wil automatically implement, and then manually implement the build method and enforce immutability there.